foodsaving/conversations/api.py

```diff
@@ -101,6 +115,52 @@ class ConversationMessageViewSet(
     def get_queryset(self):
         return self.queryset.filter(conversation__participants=self.request.user)
 
+    @detail_route(
+        methods=('POST',),
+        filter_fields=('name',),
+        permission_classes=(IsAuthenticated, IsMessageConversationParticipant)
+    )
+    def reactions(self, request, pk):
+        """route for POST /messages/{id}/reactions/ with body {"name":"emoji_name"}"""
+
+        message = get_object_or_404(ConversationMessage.objects, id=pk)
+
+        # object permissions check has to be triggered manually
+        self.check_object_permissions(self.request, message)
+
+        data = {
+            'message': pk,
+            'name': request.data['name'],
+            'user': request.user.id,
+        }
+
+        serializer = ConversationMessageReactionSerializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    @detail_route(
+        methods=('DELETE',),
+        url_path='reactions/(?P<name>[a-z0-9_+-]+)',
+        url_name='remove_reaction',
+        permission_classes=(IsAuthenticated, IsMessageConversationParticipant)
+    )
+    def remove_reaction(self, request, pk, name):
+        """route for DELETE /messages/{id}/reactions/{name}/"""
+
+        name = EmojiField.to_internal_value(None, name)
+
+        message = get_object_or_404(ConversationMessage.objects, id=pk)
```

Comment:
Optional: according to https://docs.djangoproject.com/en/2.0/_modules/django/shortcuts/#get_object_or_404 you can also pass a model to `get_object_or_404`. Makes the code a bit shorter: `message = get_object_or_404(ConversationMessage, id=pk)` (especially in the second case below)